### ISSUE TITLE
convert pth_to_hf error

### ISSUE THREAD

**Issue by bdqfork** [InternLM/xtuner#217]
![f6ea8f66976811d6314de22f67d2a07](https://github.com/InternLM/xtuner/assets/23702980/66fc4cbc-0293-4bcf-99d9-c77f45184e43)

```
File "xxx/anaconda3/envs/test/lib/python3.9/site-packages/bitsandbytes/nn/modules.py", line 158, in from_state_dict
    data = state_dict.pop(prefix.rstrip('.'))
KeyError: 'llm.base_model.model.model.layers.0.self_attn.q_proj.base_layer.weight'
```

**Comment by bdqfork** [comment]
Why key missing in lora pth?

**Comment by LZHgrla** [comment]
Could you please paste the convert command？

**Comment by bdqfork** [comment]
`convert pth_to_hf zephyr_7b_beta_qlora_alpaca_e3_copy.py work_dirs/zephyr_7b_beta_qlora_alpaca_e3_copy/iter_51000.pth models/zephy-7b-function`

**Comment by bdqfork** [comment]
> Could you please paste the convert command？

`xtuner convert pth_to_hf zephyr_7b_beta_qlora_alpaca_e3_copy.py work_dirs/zephyr_7b_beta_qlora_alpaca_e3_copy/iter_51000.pth models/zephy-7b-function`

**Comment by bdqfork** [comment]
Below is my train config

`# Copyright (c) OpenMMLab. All rights reserved.
from functools import partial
import torch
from bitsandbytes.optim import PagedAdamW32bit, AdamW
from datasets import load_dataset
from mmengine.dataset import DefaultSampler
from mmengine.hooks import (CheckpointHook, DistSamplerSeedHook, IterTimerHook,
                            LoggerHook, ParamSchedulerHook)
from mmengine.optim import AmpOptimWrapper, CosineAnnealingLR
from peft import LoraConfig
from transformers import (AutoModelForCausalLM, AutoTokenizer,
                          BitsAndBytesConfig)

from xtuner.dataset import process_hf_dataset
from xtuner.dataset.collate_fns import default_collate_fn
from xtuner.dataset.map_fns import template_map_fn_factory
from xtuner.engine import DatasetInfoHook, EvaluateChatHook
from xtuner.model import SupervisedFinetune
from xtuner.utils import PROMPT_TEMPLATE, SYSTEM_TEMPLATE

#######################################################################
#                          PART 1  Settings                           #
#######################################################################
# Model
# pretrained_model_name_or_path = 'HuggingFaceH4/zephyr-7b-beta'

pretrained_model_name_or_path = '/data/models/openbuddy-zephyr-7b-v14.1'

# Data
alpaca_en_path = 'data.json'
prompt_template = PROMPT_TEMPLATE.zephyr
max_length = 2048
pack_to_max_length = False

# Scheduler & Optimizer
batch_size = 1  # per_device
accumulative_counts = 16
dataloader_num_workers = 0
max_epochs = 1
optim_type = AdamW
lr = 2e-4
betas = (0.9, 0.999)
weight_decay = 0
max_norm = 1  # grad clip

# Evaluate the generation performance during the training
evaluation_freq = 500
SYSTEM = SYSTEM_TEMPLATE.alpaca
evaluation_inputs = [
    '请给我介绍五个上海的景点', 'Please tell me five scenic spots in Shanghai'
]

#######################################################################
#                      PART 2  Model & Tokenizer                      #
#######################################################################
tokenizer = dict(
    type=AutoTokenizer.from_pretrained,
    pretrained_model_name_or_path=pretrained_model_name_or_path,
    trust_remote_code=True,
    padding_side='right')

model = dict(
    type=SupervisedFinetune,
    llm=dict(
        type=AutoModelForCausalLM.from_pretrained,
        pretrained_model_name_or_path=pretrained_model_name_or_path,
        trust_remote_code=True,
        torch_dtype=torch.float16,
        quantization_config=dict(
            type=BitsAndBytesConfig,
            load_in_4bit=True,
            load_in_8bit=False,
            llm_int8_threshold=6.0,
            llm_int8_has_fp16_weight=False,
            bnb_4bit_compute_dtype=torch.float16,
            bnb_4bit_use_double_quant=True,
            bnb_4bit_quant_type='nf4')),
    lora=dict(
        type=LoraConfig,
        r=64,
        lora_alpha=16,
        lora_dropout=0.1,
        bias='none',
        task_type='CAUSAL_LM'))

#######################################################################
#                      PART 3  Dataset & Dataloader                   #
#######################################################################
alpaca_en = dict(
    type=process_hf_dataset,
    dataset=dict(type=load_dataset, path='json',
                 data_files=dict(train=alpaca_en_path)),
    tokenizer=tokenizer,
    max_length=max_length,
    dataset_map_fn=None,
    template_map_fn=dict(
        type=template_map_fn_factory, template=dict(
            SYSTEM='<|system|>\n{system}\n',
            INSTRUCTION='{input}\n<|assistant|>\n'),),
    remove_unused_columns=True,
    shuffle_before_pack=True,
    pack_to_max_length=pack_to_max_length)

train_dataloader = dict(
    batch_size=batch_size,
    num_workers=dataloader_num_workers,
    dataset=alpaca_en,
    sampler=dict(type=DefaultSampler, shuffle=True),
    collate_fn=dict(type=default_collate_fn))

#######################################################################
#                    PART 4  Scheduler & Optimizer                    #
#######################################################################
# optimizer
optim_wrapper = dict(
    type=AmpOptimWrapper,
    optimizer=dict(
        type=optim_type, lr=lr, betas=betas, weight_decay=weight_decay),
    clip_grad=dict(max_norm=max_norm, error_if_nonfinite=False),
    accumulative_counts=accumulative_counts,
    loss_scale='dynamic',
    dtype='float16')

# learning policy
# More information: https://github.com/open-mmlab/mmengine/blob/main/docs/en/tutorials/param_scheduler.md  # noqa: E501
param_scheduler = dict(
    type=CosineAnnealingLR,
    eta_min=lr * 0.1,
    by_epoch=True,
    T_max=max_epochs,
    convert_to_iter_based=True)

# train, val, test setting
train_cfg = dict(by_epoch=True, max_epochs=max_epochs, val_interval=1)

#######################################################################
#                           PART 5  Runtime                           #
#######################################################################
# Log the dialogue periodically during the training process, optional
custom_hooks = [
    dict(type=DatasetInfoHook, tokenizer=tokenizer),
    dict(
        type=EvaluateChatHook,
        tokenizer=tokenizer,
        every_n_iters=evaluation_freq,
        evaluation_inputs=evaluation_inputs,
        system=SYSTEM,
        prompt_template=prompt_template)
]

# configure default hooks
default_hooks = dict(
    # record the time of every iteration.
    timer=dict(type=IterTimerHook),
    # print log every 100 iterations.
    logger=dict(type=LoggerHook, interval=10),
    # enable the parameter scheduler.
    param_scheduler=dict(type=ParamSchedulerHook),
    # save checkpoint per epoch.
    checkpoint=dict(type=CheckpointHook, interval=1000, by_epoch=False),
    # set sampler seed in distributed evrionment.
    sampler_seed=dict(type=DistSamplerSeedHook),
)

# configure environment
env_cfg = dict(
    # whether to enable cudnn benchmark
    cudnn_benchmark=False,
    # set multi process parameters
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0),
    # set distributed parameters
    dist_cfg=dict(backend='nccl'),
)

# set visualizer
visualizer = None

# set log level
log_level = 'INFO'

# load from which checkpoint
load_from = None

# whether to resume training from the loaded checkpoint
resume = False

# Defaults to use random seed and disable `deterministic`
randomness = dict(seed=None, deterministic=False)`

**Comment by LZHgrla** [comment]
@bdqfork 
What is the version of `bitsandbytes` in your env?

**Comment by bdqfork** [comment]
> bitsandbytes

0.41.2.post2

**Comment by LZHgrla** [comment]
@bdqfork 
I repro this error!
It seems to be a side effect of https://github.com/TimDettmers/bitsandbytes/pull/753. This has been fix in https://github.com/TimDettmers/bitsandbytes/pull/864

You can run `pip install -U bitsandbytes` to fix it!

**Comment by bdqfork** [comment]
> @bdqfork I repro this error! It seems to be a side effect of [TimDettmers/bitsandbytes#753](https://github.com/TimDettmers/bitsandbytes/pull/753). This has been fix in [TimDettmers/bitsandbytes#864](https://github.com/TimDettmers/bitsandbytes/pull/864)
> 
> You can run `pip install -U bitsandbytes` to fix it!

 I will try later! Thanks!

**Comment by bdqfork** [comment]
After reinstalling xtuner and updating bitsandbytes, it's fixed, thanks!
